src/doom/am_map.c

```diff
@@ -1164,6 +1201,17 @@ AM_drawFline
     
     static int fuck = 0;
 
+    if (crispy->smoothmap) // [crispy] Use DrawWuLine from Heretic
```

Comment:
Here it is getting beefy. I want to avoid this check being evaluated for each line during each tic. Could you please factor this out into two separate line drawing functions, e.g. `AM_drawFline_Vanilla()` for the current implementation and `EM_drawWuLine()`for the smoothed approach, and then introduce a function pointer `AM_drawFline()` that you set to either of them in `AM_LevelInit()` depending on the value of `crispy->smoothmap`. 

Then you'll have to run `AM_LevelInit(true)` from `M_CrispyToggleSmoothMap()`, of course.

Please don't hesitate to ask if this point is not clear to you.